engine/src/main/java/org/terasology/engine/module/rendering/RenderingModuleRegistry.java

```diff
@@ -0,0 +1,163 @@
+     */
+    public List<ModuleRendering> updateRenderingModulesOrder(ModuleEnvironment moduleEnvironment, Context context) {
+        this.orderedModuleRenderingInstances = calculateModuleOrder(fetchRenderingModules(moduleEnvironment, context));
+        return this.orderedModuleRenderingInstances;
+    }
+
+    /**
+     * Sets new context for each ModuleRendering instance
+     * @param context
+     */
+    public void updateModulesContext(Context context) {
+        for (ModuleRendering moduleRendering : orderedModuleRenderingInstances) {
+            moduleRendering.setContext(context);
+            // context.put(ModuleRendering.class, moduleRendering);
+        }
+    }
+
+    public void setRenderingModulePriority(ModuleRendering module, int initializationPriority) {
+        module.setInitPriority(initializationPriority);
+    }
+
+    public int getRenderingModulePriority(ModuleRendering module) {
+        return module.getInitPriority();
+    }
+
+    public Name getRenderingModuleId(ModuleRendering module) {
+        return module.getProvidingModule();
+    }
+
+    /**
+     * Sorts given list of {@link ModuleRendering} instances by {@code ModuleRendering.initializationPriority} attribute, saves and returns it.
+     * @param activeModuleRenderingInstances List of {@link ModuleRendering} instances
+     * @return Ordered list of {@link ModuleRendering} instances
+     */
+    private List<ModuleRendering> calculateModuleOrder(List<ModuleRendering> activeModuleRenderingInstances) {
+        // Sorted copy of activeRenderingModules
+        orderedModuleRenderingInstances = activeModuleRenderingInstances.stream()
+                .sorted(Comparator.comparing(ModuleRendering::getInitPriority)) // sort ascending by initializationPriority attribute
+                .collect(Collectors.toList()); //convert stream to List again
+        return orderedModuleRenderingInstances;
+    }
+
+    /**
+     * Fetches all loaded {@link ModuleRendering} classes from the given {@link ModuleEnvironment}
+     * and instantiatest them with current context.
+     * @param moduleEnvironment
+     * @param context
+     * @return list of {@link ModuleRendering} instances
+     */
+    private List<ModuleRendering> fetchRenderingModules(ModuleEnvironment moduleEnvironment, Context context) {
+        List<ModuleRendering> moduleList = new ArrayList<>();
+
+        for (Class<? extends ModuleRendering> renderingClass : moduleEnvironment.getSubtypesOf(ModuleRendering.class)) {
+            ModuleRendering moduleRenderingInstance = null;
+            for (ModuleRendering moduleRendering : orderedModuleRenderingInstances) {
+                if (renderingClass.isInstance(moduleRendering)) {
+                    moduleRenderingInstance = moduleRendering;
+                    // TODO there's marginal scenarios where you could have more instances of same class moduleRendering
+                    if (!moduleList.contains(moduleRenderingInstance)) {
```

Comment:
Is `moduleList` an ordered list? Otherwise, we could use a `Set` here and get rid of the `contains(...)` check.